src/Features/LanguageServer/Protocol/Handler/Rename/RenameHandler.cs

```diff
@@ -2,43 +2,35 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Editor;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
-using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
 
-namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
-    internal class RenameHandler : ILspRequestHandler<RenameParams, WorkspaceEdit, Solution>
+    [ExportLspMethod(LSP.Methods.TextDocumentRenameName), Shared]
+    internal class RenameHandler : IRequestHandler<LSP.RenameParams, WorkspaceEdit>
     {
-        private readonly IThreadingContext _threadingContext;
-
+        [ImportingConstructor]
         public RenameHandler(IThreadingContext threadingContext)
```

Comment:
I think you can delete the constructor all together since we no longer need the parameter.  MEF should be able to create the type with the default parameter-less constructor

Reply:
Hmm, I tried deleting it and received the following error: "'RenameHandler' is MEF-exported and should have a single importing constructor of the correct form".
Not sure if I need to make other modifications, but I've added it back for now.

Reply:
Hmm strange, could be a mef cache issue or something similar.  It shouldn't be needed, see http://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/FoldingRanges/FoldingRangesHandler.cs is exported w/out a constructor.

Reply:
Oh interesting... is the difference perhaps that the rename constructor takes in a parameter? I'm not sure, but the error message above was from Azure DevOps, so would the mef cache still potentially be an issue there?

Reply:
If you have a constructor with a parameter you'll need the importing constructor annotation, but we should be able to use the parameterless default constructor which shouldn't need an annotation.  

Idt mef cache issue should be in CI - could you point me to the one that failed?

Reply:
Here's the run that failed: https://github.com/dotnet/roslyn/runs/488960603

Reply:
Ah well that error is coming from here - https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructor.cs#L107

@sharwell is this rule new?  Is it required that a mef exported type with the default parameter-less constructor implement a parameter-less constructor and add the importingconstructor attribute?

Reply:
@dibarbet Yes the rule is new. It was the first of a two-step process to enable analysis of the standard MEF construction form:

```csharp
[ImportingConstructor]
[Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
public TypeName()
{
}
```

Analysis of the `[Obsolete]` attribute is next.

Reply:
I see, good to know, thanks!  

@allisonchou my preference would be then implementing it as Sam suggested above without the threading context, but it's fine to leave it as-is for now and take care of it in a followup if you have time.